sim_pkg/config.py

```diff
@@ -0,0 +1,100 @@
+from dataclasses import dataclass, field
```

Comment:
You're free to disagree on this solution but I do think it's much cleaner to have a `Configuration` class that encapsulates all your -- well -- configuration. It's easier to access fields like this (autocompletion), you have cleaner scope, and it all around feels better to use than those global variables.

We're writing software here, not scripts, make it look like software! :)

Reply:
Thanks a lot for the detailed comments. I have gone through and accepted most of the comments. Just trying to fix the errors from config and hopefully get this running. I shall let you know once I am able to get it to run.